airflow/models.py

```diff
@@ -910,16 +910,34 @@ def is_queueable(
         elif self.state == State.QUEUED and not include_queued:
             return False
         # is the task runnable and have its dependencies been met?
-        elif (
-                self.state in State.runnable() and
-                self.are_dependencies_met(
-                    ignore_depends_on_past=ignore_depends_on_past,
-                    flag_upstream_failed=flag_upstream_failed)):
+        elif (self.state in State.runnable() and
+                  self.are_dependencies_met(
+                      ignore_depends_on_past=ignore_depends_on_past,
+                      flag_upstream_failed=flag_upstream_failed) and
+                  not self.is_superceded_by_future()):
             return True
         # anything else
         else:
             return False
 
+    @provide_session
+    def is_superceded_by_future(self, session):
+        if not self.task.only_run_latest:
+            return False
+        logging.info("checking for future")
```

Comment:
Not sure how much value this log line provides. How verbose is it going to be? If you elect to keep it, having more information like the name of the task would be useful.